pkg/codegen/go/gen.go

```diff
@@ -3526,7 +3531,7 @@ func GeneratePackage(tool string, pkg *schema.Package) (map[string][]byte, error
 				packageRegex = fmt.Sprintf("^%s(/v\\d+)?", pkg.importBasePath)
 			}
 
-			_, err := fmt.Fprintf(buffer, utilitiesFile, packageRegex)
+			_, err := fmt.Fprintf(buffer, UtilitiesFile(pkg), packageRegex)
```

Comment:
Do we need to make sure `pkg.needsUtils` is set to true if there are any resources or invokes? Otherwise, is it possible the utilities file won't be emitted?

Reply:
`insertPkgDefaultsOptsCall` sets `pkg.needsUtils = true`.